[stdlib] Implement `Dict.popitem()` method

msaelices opened the pull request:
- [X] Implement `Dict.popitem()` method
- [X] Added changelog entry

msaelices committed:
[stdlib] Implement Dict.popitem() method

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
Merge branch 'nightly' into dict-popitem

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
[stdlib] Better error message if Dict.popitem() is empty

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
[stdlib] Remove duplicated Dict.__iter__() method

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
[stdlib] Changelog entry for Dict.popitem()

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

JoeLoser reviewed (approved):
Looks good — just one minor wording suggestion on the changelog entry, and then I'm happy to import and merge this.

msaelices committed:
[stdlib] Honours the Python dict.popitem() LIFO ordering

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
Minor wording change on changelog's Dict.popitem() entry

Co-authored-by: Joe Loser <joeloser@fastmail.com>
Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
[stdlib] Fix changelog.md line width

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
Merge branch 'nightly' into dict-popitem

msaelices committed:
[stdlib] Refactor Dict.popitem() code to prevent fatal errors.
Fatal errors were detected if we add a new key after calling popitem()
Thanks to @jayzhan211 for the heads-up

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
[stdlib] Add @jayzhan211 to the changelog's entry attribution

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

msaelices committed:
[stdlib] Remove testing code

Signed-off-by: Manuel Saelices <msaelices@gmail.com>

jayzhan211 reviewed (approved):
👍
